### DISCUSSION TITLE
CallerMemberName not working with lambda default values?

### DISCUSSION THREAD

**Discussion by marius-bughiu** [dotnet/csharplang#7267] [General]
Was playing with the default values feature for lambdas, and noticed `CallerMemberName` is not working. Is it a bug, or a preview limitation?

Can be tested easily with this:

```
using System.Runtime.CompilerServices;

var myMethod1 = ([CallerMemberName] string? param = null) => Console.WriteLine(param);
void myMethod2([CallerMemberName] string? param = null) => Console.WriteLine(param);

myMethod1();
myMethod2();
```

The call to `myMethod1` doesn't output anything.
The call to `myMethod2` correctly outputs `<Main>$`.

Thoughts?

**Comment by CyrusNajmabadi** [comment]
This is by design. The delegates created are System.Func and Action, nether of which have that attribute on them. 

Use a custom delegate type with that attribute of if you want this behavior. Thanks!
